hoodie-client/src/main/java/com/uber/hoodie/index/hbase/HBaseIndex.java

```diff
@@ -103,48 +132,67 @@ private Connection getHBaseConnection() {
 
     private final HoodieTable<T> hoodieTable;
 
+    private Integer multiGetBatchSize = Integer.parseInt(config.getProps()
+        .getProperty(HoodieIndexConfig.HBASE_READ_BATCH_SIZE_PROP));
+
     LocationTagFunction(HoodieTable<T> hoodieTable) {
       this.hoodieTable = hoodieTable;
     }
 
     @Override
     public Iterator<HoodieRecord<T>> call(Integer partitionNum,
-        Iterator<HoodieRecord<T>> hoodieRecordIterator) {
+                                          Iterator<HoodieRecord<T>> hoodieRecordIterator) {
       // Grab the global HBase connection
       synchronized (HBaseIndex.class) {
-        if (hbaseConnection == null) {
+        if (hbaseConnection == null || hbaseConnection.isClosed()) {
           hbaseConnection = getHBaseConnection();
         }
       }
       List<HoodieRecord<T>> taggedRecords = new ArrayList<>();
       HTable hTable = null;
       try {
         hTable = (HTable) hbaseConnection.getTable(TableName.valueOf(tableName));
+        List<Get> statements = new ArrayList<>();
+        List<HoodieRecord> currentBatchOfRecords = new LinkedList<>();
         // Do the tagging.
         while (hoodieRecordIterator.hasNext()) {
           HoodieRecord rec = hoodieRecordIterator.next();
-          // TODO(vc): This may need to be a multi get.
-          Result result = hTable.get(
-              new Get(Bytes.toBytes(rec.getRecordKey())).setMaxVersions(1)
-                  .addColumn(SYSTEM_COLUMN_FAMILY, COMMIT_TS_COLUMN)
-                  .addColumn(SYSTEM_COLUMN_FAMILY, FILE_NAME_COLUMN)
-                  .addColumn(SYSTEM_COLUMN_FAMILY, PARTITION_PATH_COLUMN));
-
-          // first, attempt to grab location from HBase
-          if (result.getRow() != null) {
-            String commitTs =
-                Bytes.toString(result.getValue(SYSTEM_COLUMN_FAMILY, COMMIT_TS_COLUMN));
-            String fileId =
-                Bytes.toString(result.getValue(SYSTEM_COLUMN_FAMILY, FILE_NAME_COLUMN));
-
-            HoodieTimeline commitTimeline = hoodieTable.getCompletedCommitTimeline();
-            // if the last commit ts for this row is less than the system commit ts
-            if (!commitTimeline.empty() && commitTimeline.containsInstant(
-                new HoodieInstant(false, HoodieTimeline.COMMIT_ACTION, commitTs))) {
-              rec.setCurrentLocation(new HoodieRecordLocation(commitTs, fileId));
+          statements.add(generateStatement(rec.getRecordKey()));
+          currentBatchOfRecords.add(rec);
+          // iterator till we reach batch size
+          if (statements.size() < multiGetBatchSize && hoodieRecordIterator.hasNext()) {
```

Comment:
Good use of continue to handle leftover records. 

can we tweak this to make the actual code fit into if block without use of continue.. might be better for readability..